disnake/types/interactions.py

```diff
@@ -202,40 +202,40 @@ class ModalInteractionData(TypedDict):
     components: List[ActionRow]
 
 
-InteractionData = Union[
-    ApplicationCommandInteractionData, ComponentInteractionData, ModalInteractionData
-]
-
-
-class BaseInteraction(TypedDict):
-    id: Snowflake
-    application_id: Snowflake
-    type: InteractionType
-    token: str
-    version: int
-
-
 # common properties in appcmd and component interactions (or generally, non-ping interactions)
-class _InteractionOptional(TypedDict, total=False):
+class _BaseInteractionOptional(TypedDict, total=False):
     guild_id: Snowflake
     # one of these two will always exist, according to docs
     user: User
     member: MemberWithUser
     guild_locale: str
 
 
-class Interaction(BaseInteraction, _InteractionOptional):
+class _BaseInteraction(_BaseInteractionOptional):
+    id: Snowflake
+    application_id: Snowflake
+    token: str
+    version: int
+
+
+class _BaseUserInteraction(_BaseInteraction):
     # the docs specify `channel_id` as optional,
-    # but it is assumed to always exist on appcmd and component interactions
+    # but it is assumed to always exist on non-ping interactions
     channel_id: Snowflake
     locale: str
 
 
-class ApplicationCommandInteraction(Interaction):
+class PingInteraction(_BaseInteraction):
```

Comment:
This does not include _BaseInteractionOptional ever, according to the docs. Does it still make sense to inherit from that TypedDict?

Reply:
Oh, good catch, it should never have any of those fields. Fixed in b7a865123da98d0e4290d5011c07c27524459ac9